client/components/modals/FileUploadModal/FileUploadModal.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+	Modal,
+	Box,
+	Field,
+	FieldGroup,
+	TextInput,
+	ButtonGroup,
+	Button,
+} from '@rocket.chat/fuselage';
+import { useAutoFocus } from '@rocket.chat/fuselage-hooks';
+import React, {
+	ReactElement,
+	memo,
+	useState,
+	ChangeEvent,
+	FormEventHandler,
+	useEffect,
+} from 'react';
+
+import { useToastMessageDispatch } from '../../../contexts/ToastMessagesContext';
+import { useTranslation } from '../../../contexts/TranslationContext';
+import FilePreview from './FilePreview';
+
+type FilePreviewModalProps = {
+	onClose: () => void;
+	onSubmit: (name: string, description?: string) => void;
+	file: File;
+	isValidContentType: boolean;
+};
+
+const FilePreviewModal = ({
+	onClose,
+	file,
+	onSubmit,
+	isValidContentType,
+}: FilePreviewModalProps): ReactElement => {
+	const [name, setName] = useState<string>(file.name);
+	const [description, setDescription] = useState<string>('');
+	const t = useTranslation();
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const ref = useAutoFocus();
+
+	const handleName = (e: ChangeEvent<HTMLInputElement>): void => {
+		setName(e.currentTarget.value);
+	};
+
+	const handleDescription = (e: ChangeEvent<HTMLInputElement>): void => {
+		setDescription(e.currentTarget.value);
+	};
+
+	const handleSubmit: FormEventHandler<HTMLFormElement> = (e): void => {
+		e.preventDefault();
+		if (!name) {
+			return dispatchToastMessage({
+				type: 'success',
+				message: t('error-the-field-is-required', { field: t('Name') }),
+			});
+		}
+		onSubmit(name, description);
+	};
+
+	useEffect(() => {
+		if (!isValidContentType) {
+			dispatchToastMessage({
+				type: 'error',
+				message: t('FileUpload_MediaType_NotAccepted__type__', { type: file.type }),
+			});
+			onClose();
+			return;
+		}
+
+		if (file.size === 0) {
+			dispatchToastMessage({
+				type: 'error',
+				message: t('FileUpload_File_Empty'),
+			});
+			onClose();
+		}
+	}, [file, dispatchToastMessage, isValidContentType, t, onClose]);
+
+	return (
+		<Modal>
```

Comment:
Why not the `GenericModal`?

Reply:
In order to enable confirming upload with the `Enter` key, I had to place a form in the component a modify a button to be of `type='submit'`.

Reply:
In your opinion, shouldn't we have this behavior in the `GenericModal`?